pkg/virt-handler/vm.go

```diff
@@ -1442,6 +1442,21 @@ func (c *VirtualMachineController) sync(key string,
 		}
 	}
 
+	if (shouldShutdown || forceShutdownIrrecoverable || shouldDelete) && domainExists {
+		if vmi.Spec.TerminationGracePeriodSeconds != nil {
+			gracePeriod := *vmi.Spec.TerminationGracePeriodSeconds
+
+			if domain.Spec.Metadata.KubeVirt.GracePeriod == nil {
+				domain.Spec.Metadata.KubeVirt.GracePeriod = &api.GracePeriodMetadata{}
+			}
+
+			if domain.Spec.Metadata.KubeVirt.GracePeriod.DeletionGracePeriodSeconds != gracePeriod {
+				log.Log.Object(vmi).Infof("Syncing GracePeriod: setting DeletionGracePeriodSeconds to %d", gracePeriod)
+				domain.Spec.Metadata.KubeVirt.GracePeriod.DeletionGracePeriodSeconds = gracePeriod
```

Comment:
This would modify a cached value and would be overridden on next notification from launcher. We need to treat this where needed, best with some kind of helper function